### PR TITLE
System tests: quadlet network test

### PR DESCRIPTION
**Pull request by ygalblum**
#### Does this PR introduce a user-facing change?
No

```release-note
None
```

### CODE REVIEW

File: test/system/252-quadlet.bats [hunk 1]
```diff
@@ -283,4 +283,66 @@ EOF
...
+
+    service_cleanup $QUADLET_SERVICE_NAME inactive
+}
+
+# A quadlet container depends on a quadlet volume
+@test "quadlet - network dependency" {
+    local quadlet_network_file=$PODMAN_TMPDIR/dep_$(random_string).network
+    cat > $quadlet_network_file <<EOF
+[Volume]
+EOF
```
Comment (vrothberg) [inline test/system/252-quadlet.bats:314]: I am confused by the `.network` suffix for a VOLUME file.

File: test/system/252-quadlet.bats [hunk 2]
```diff
@@ -283,4 +283,66 @@ EOF
...
+    cat > $quadlet_file <<EOF
+[Container]
+Image=$IMAGE
+Exec=top
+Network=$network_service
+EOF
+
+    # Volume should not exist
+    run_podman network ls
+    assert "$output" !~ ".*${network_name}.*"
```
Comment (vrothberg) [inline test/system/252-quadlet.bats:331]: Let's use `network exists` for that check.

File: test/system/252-quadlet.bats [hunk 3]
```diff
@@ -283,4 +283,66 @@ EOF
...
+
+    local quadlet_file=$PODMAN_TMPDIR/user_$(random_string).container
+    cat > $quadlet_file <<EOF
+[Container]
+Image=$IMAGE
+Exec=top
+Network=$network_service
+EOF
+
+    # Volume should not exist
```
Comment (vrothberg) [inline test/system/252-quadlet.bats:329]: Volume or network?

**Review by vrothberg — COMMENTED** [review]
Looks like there are copy-paste artifacts from the volume test left.

File: test/system/252-quadlet.bats (hunk 1 above)
Comment (ygalblum) [reply test/system/252-quadlet.bats:313]: Fixed

File: test/system/252-quadlet.bats (hunk 2 above)
Comment (ygalblum) [reply test/system/252-quadlet.bats]: Fixed

File: test/system/252-quadlet.bats (hunk 3 above)
Comment (ygalblum) [reply test/system/252-quadlet.bats]: Fixed

**Comment by ygalblum** [comment]
> Looks like there are copy-paste artifacts from the volume test left.

@vrothberg Sorry for these. There're now fixed

File: test/system/252-quadlet.bats [hunk 4]
```diff
@@ -283,6 +283,62 @@ EOF
...
+
+    service_setup $QUADLET_SERVICE_NAME
+
+    # Network system unit should be active
+    run systemctl show --property=ActiveState "$network_service"
+    assert "$output" = "ActiveState=active" \
+           "network should be active via dependency"
+
+    # Network should exist
+    run_podman network exists $network_name
```
Comment (vrothberg) [inline test/system/252-quadlet.bats:339]: Can we clean up the container and network after?

Comment (vrothberg) [reply test/system/252-quadlet.bats:339]: Apologies for not catching that before.

**Comment by ygalblum** [comment]
@vrothberg following your last comment I realized that there was something wrong in the test itself. This lead me to find an issue in the Volume test (copy pasting erroneous code will produce erroneous code :) ). The fix to the Volume test is here: #16970.
Once both tests were fixed I found an issue in the handling of Network services in Quadlet. The fix is here: #16971. 
So, while the test here is correct, it will only pass after #16971 is merged and this branch is rebased on it

**Comment by ygalblum** [comment]
The fix to the code (#16971) was merged to main. This branch was rebased on top of the new main. So, it's ready to merge

File: test/system/252-quadlet.bats (hunk 4 above)
Comment (ygalblum) [reply test/system/252-quadlet.bats:358]: Done

**Comment by rhatdan** [comment]
LGTM
@edsantiago @vrothberg PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM
but would like head nods from @edsantiago and @vrothberg

**Pull request merged**